[Feat] Add info icon to dashboard title (#1053)
mckinsey/vizro: vizro-core/changelog.d/20250310_162006_nadija_ratkusic_graca_add_info_icon.md

<!--
A new scriv changelog fragment.

Uncomment the section that is right (remove the HTML comment wrapper).
-->

<!--
### Highlights ✨

- A bullet item for the Highlights ✨ category with a link to the relevant PR at the end of your entry, e.g. Enable feature XXX. ([#1](https://github.com/mckinsey/vizro/pull/1))

-->
<!--
### Removed

- A bullet item for the Removed category with a link to the relevant PR at the end of your entry, e.g. Enable feature XXX. ([#1](https://github.com/mckinsey/vizro/pull/1))

-->
### Added

> - Added `Title` model to use within `vm.Dashboard`, enabling support for info icon and tooltip in the dashboard header. ([#1053](https://github.com/mckinsey/vizro/pull/1053))

Reviewer: ```suggestion
- Add `Title` model to use within `vm.Dashboard`, enabling support for an info icon with tooltip in the dashboard header. ([#1053](https://github.com/mckinsey/vizro/pull/1053))
```
